discovery/src/server.rs

```diff
@@ -0,0 +1,236 @@
+use std::borrow::Cow;
+use std::collections::BTreeMap;
+use std::convert::Infallible;
+use std::net::{Ipv4Addr, SocketAddr};
+use std::pin::Pin;
+use std::sync::Arc;
+use std::task::{Context, Poll};
+
+use aes_ctr::cipher::generic_array::GenericArray;
+use aes_ctr::cipher::{NewStreamCipher, SyncStreamCipher};
+use aes_ctr::Aes128Ctr;
+use futures_core::Stream;
+use hmac::{Hmac, Mac, NewMac};
+use hyper::service::{make_service_fn, service_fn};
+use hyper::{Body, Method, Request, Response, StatusCode};
+use log::{debug, warn};
+use serde_json::json;
+use sha1::{Digest, Sha1};
+use tokio::sync::{mpsc, oneshot};
+
+use crate::core::authentication::Credentials;
+use crate::core::config::DeviceType;
+use crate::core::diffie_hellman::DhLocalKeys;
+
+type Params<'a> = BTreeMap<Cow<'a, str>, Cow<'a, str>>;
+
+pub struct Config {
+    pub name: Cow<'static, str>,
+    pub device_type: DeviceType,
+    pub device_id: String,
+}
+
+struct RequestHandler {
+    config: Config,
+    keys: DhLocalKeys,
+    tx: mpsc::UnboundedSender<Credentials>,
+}
+
+impl RequestHandler {
+    fn new(config: Config) -> (Self, mpsc::UnboundedReceiver<Credentials>) {
+        let (tx, rx) = mpsc::unbounded_channel();
+
+        let discovery = Self {
+            config,
+            keys: DhLocalKeys::random(&mut rand::thread_rng()),
+            tx,
+        };
+
+        (discovery, rx)
+    }
+
+    fn handle_get_info(&self) -> Response<hyper::Body> {
+        let public_key = base64::encode(&self.keys.public_key());
+        let device_type: &str = self.config.device_type.into();
+
+        let body = json!({
+            "status": 101,
+            "statusString": "ERROR-OK",
+            "spotifyError": 0,
+            "version": "2.7.1",
+            "deviceID": (self.config.device_id),
+            "remoteName": (self.config.name),
+            "activeUser": "",
+            "publicKey": (public_key),
+            "deviceType": (device_type),
+            "libraryVersion": "0.1.0",
```

Comment:
Should this not be the `librespot` version?

Reply:
I don't know, it was so before. But why should the official Spotify clients need the librespot version?

Reply:
I guess the same like HTTP servers shouldn't need the user agent, or the HTTP client the server signature, except to work around quirks. Googling around I see that official Connect clients set a variety of versions, which seem to correspond to their internal software rather than some Spotify-supplied spec.

Reply:
```suggestion
            "libraryVersion": crate::core::version::SEMVER,
```
I could try this, let's see if it compiles.